showcase/scripts/verify.sh

```diff
@@ -1,70 +1,27 @@
 #!/bin/sh
-# This script is executed by ../BUILD.bazel as a final post-generation step.
+# This script generates showcase in a temporary/untracked folder and compares
+# its contents with the actual showcase libraries. 
 
 set -o errexit
-BAZEL_ROOT=$PWD
-SHOWCASE_DIR="$BUILD_WORKSPACE_DIRECTORY/showcase"
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source "${SCRIPT_DIR}/showcase_utilities.sh"
+SHOWCASE_DIR="${SCRIPT_DIR}/.."
 
-clear_existing() {
-  cd "$SHOWCASE_DIR/$1"
-  find . -name '*.java' -not -path '*/it/*' -delete
-  find . -name 'gapic_metadata.json' -delete
-  find . -name 'reflect-config.json' -delete
-  cd -
-}
-create_unpack_dir() {
-  cd "$BAZEL_ROOT"
-  rm -rf "$1"
-  mkdir "$1"
-  cd "$1"
-}
-delete_unneeded() {
-  find . -name '.DS_Store' -delete
-  find . -name 'PlaceholderFile.java' -delete
-  find . -type d -empty -delete
-}
+# generate sources
+readonly REPLACE_SOURCE='false'
+readonly PERFORM_CLEANUP='false'
+bash "${SCRIPT_DIR}/generate_components.sh" "${REPLACE_SOURCE}" "${PERFORM_CLEANUP}"
 
-case $1 in
-  proto)
-    PROTO_PROJECT_DIR=proto-gapic-showcase-v1beta1
-    PROTO_ARCHIVE_NAME=proto-google-cloud-showcase-v1beta1-java
-    PROTO_ARCHIVE=$(find . -name "$PROTO_ARCHIVE_NAME.tar.gz")
-    create_unpack_dir proto_unpacked
-    PROTO_UNPACK_DIR=$PWD
+# compare with proto library
+PROTO_PROJECT_DIR='proto-gapic-showcase-v1beta1'
+diff -ru "${SHOWCASE_DIR}/${PROTO_PROJECT_DIR}/src" "${SCRIPT_DIR}/showcase-output/proto-showcase-output/src"
```

Comment:
Now that we are generating all three modules at the same time, is it possible to have a diff for all three modules?

Reply:
Unless there is a better way than preparing two special folders with a similar structure and run `diff` on their roots I think the simplest is to just run `diff` one time for each of the folders (`proto`, `gapic` and `grpc`) in a single call to `verify.sh`. We don't compare all the contents of `gapic-showcase` or `grpc-gapic-showcase-v1beta1` because there are other special files not generated by the scripts (e.g. ITs).